dotCMS/src/integration-test/java/com/dotcms/publishing/listener/SecurityKeyResetTest.java

```diff
@@ -0,0 +1,82 @@
+package com.dotcms.publishing.listener;
+
+import static com.dotcms.publisher.business.PublisherTestUtil.createEndpoint;
+import static com.dotcms.publisher.business.PublisherTestUtil.createEnvironment;
+
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.LicenseTestUtil;
+import com.dotcms.company.CompanyAPI;
+import com.dotcms.company.CompanyAPIFactory;
+import com.dotcms.datagen.TestUserUtils;
+import com.dotcms.publisher.endpoint.bean.PublishingEndPoint;
+import com.dotcms.publisher.endpoint.business.PublishingEndPointAPI;
+import com.dotcms.publisher.environment.bean.Environment;
+import com.dotcms.publisher.pusher.PushPublisher;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.business.APILocator;
+import com.liferay.portal.model.Company;
+import com.liferay.portal.model.User;
+import com.liferay.util.Encryptor;
+import java.security.Key;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class SecurityKeyResetTest extends IntegrationTestBase {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+        LicenseTestUtil.getLicense();
+    }
+
+    /**
+     * Given scenario: This basically tests that after a key reset the push publish endpoint are
+     * still usable for decrypting important stuff using the new key; Expected: The endpoints that
+     * were using the key that got re-generated remain usable.
+     */
+    @Test
+    public void Test_Push_Publish_Endpoints_Have_Valid_Keys_After_Key_Reset_Expect_Success() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5677177)